### ISSUE TITLE
possibility for accessibility?

### ISSUE THREAD

**Issue by lilmike** [gantry/gantry5#673]
Hi,
I was testing out the dev release of the wordpress gantry framework, and found that, unfortunately, it was not very accessible in the back end (using wordpress 4.3rc1 and hydrogen). There were quite a few unlabelled buttons and links (so many, in fact, that it was hard to go anywhere in the administration panels for me, using a screen reader). I would guess the dragging and dropping would not be accessible (by its very nature it probably wouldn't be), though I was unable to test this because I was unable to get to parts of the UI needed.
Is there any way it could be made accessible for those using screen readers? I would be more than happy to test (and even contribute code if possible) as I'd really like to use this in my own themes, and the more accessible tools we have, the better. I would guess there would have to be something done similar to wordpress widgets for accessibility (as it involves dragging and dropping) where there's a special mode for screen reader users where they can enter in positions manually (for example in the wordpress widgets you can choose a position from 1 which is first widget down to x+1 which is last widget after x widgets before it).
Anyway, I'd really like to hear your opinions on this and wouldn't mind putting in some work to help make this a reality if at all possible.
Thanks,
-Michael.

**Comment by newkind** [comment]
Hi,

We can definitely work on the button labels. About the position numbering, I'm affraid that this is also possible because of the new wordpress widget mechanism that allows you to click on a widget and select  (whout drag-and-drop) a widget position below it. This is something that isn't possible in the G5 Layout Manager. A workaround for that can be to prepare layouts and use the Widget Position particle instead of the Widget - this way you will get multiple widget positions on the widgets page and you can use them with screen reader like on any other WordPress setup.

Thoughts @mahagr ?

**Comment by lilmike** [comment]
Hi,
When I said how wordpress does it, I meant like they have an 
accessibility mode, in which the widgets no longer are dragged and 
dropped, but instead you can click to add a widget underneath each 
widget type, then choose a sidebar and position on that sidebar (from 1 
to x+1) where x is the current number of widgets on that sidebar.
Thanks,
-Michael.

**Comment by newkind** [comment]
I had the same on my mind. Just like I said, right now you can prepare the widget positions in Gantry
Layout Manager, and then use the widgets like you’d normally do. I can see what can we do with labels
after we get the first version of Gantry for WP out.


On 2 sierpnia 2015 at 14:33:18, Michael Taboada (notifications@github.com) wrote:

Hi,
When I said how wordpress does it, I meant like they have an
accessibility mode, in which the widgets no longer are dragged and
dropped, but instead you can click to add a widget underneath each
widget type, then choose a sidebar and position on that sidebar (from 1
to x+1) where x is the current number of widgets on that sidebar.
Thanks,
-Michael.

**Comment by lilmike** [comment]
Hi,
Ah. I didn't realize there was a different way of doing it. And when you 
say widgets, I'm assuming you're talking about particles? Basically I'm 
just looking for an accessible way of laying out your site in gantry.
Thanks,
-Michael.

**Comment by mahagr** [comment]
Accessibility is really something that we need to have, but I have no idea on how to make a page accessible (or best practices) as I have no experience on screen readers.

Layout and menu managers likely need some work, but all the other pages should be pretty trivial to optimize for screen readers. So we could start from there.

So in short, are there any good docs on how to make it screen reader friendly?

**Comment by lilmike** [comment]
Hi,
One resource I've heard mentioned quite a bit is WebAim (webaim.org). 
There's also a couple other sites scattered about the net that you could 
probably find with a google search for 'Guide to making a website screen 
reader friendly' or similar. In general though, webaim does have quite a 
good amount of content and where it lacks I'm sure it links to other places.
Hth,
-Michael.
P.S. I'd be willing to test any modifications you might make as it would 
be easier for me to tell if something would be a problem as I have 
experience with screen readers. Though if you needed to test, say a 
pre-release version you could always download the NVDA screen reader and 
try with that.

**Comment by marktaylor46** [comment]
There are tools available to assess web accessibility and offer suggestions - here is one such tool http://wave.webaim.org/

It might be difficult to use this on admin - but i'm sure there's a way.

I hope that helps.

**Comment by mahagr** [comment]
@lilmike Can you take a look on current CI builds if they help at all?

Any further feedback is highly appreciated, thanks!

**Comment by lilmike** [comment]
Hi,
It looks alot better in the hydrogen settings, though I wasn't able to 
test the layout creation because I couldn't find where it was. I looked 
under appearance, but it didn't seem to be there. Anyway, so far, so 
good :-).
-Michael.

**Comment by mahagr** [comment]
Layout presets (meaning sections, rows and columns) cannot be created from UI, you need to use simple YAML format to create your own custom layouts. We have some plans on making it easier in the future. Instructions on how to do that can be found from our documentation:

http://docs.gantry.org/gantry5/advanced/creating-layout-presets

If you mean creating Joomla layouts, you can just do whatever you did before and use Joomla template manager to do it.

**Comment by w00fz** [comment]
If you meant the Layout Manager, you can only access it from an Outline that is not `Base Outline`. In `Base Outline` we are hiding that tab.

**Comment by mahagr** [comment]
Moved this issue into 5.0.2 milestone as it doesn't only affect WP and the implementation has already started.

**Comment by w00fz** [comment]
Closing this. A lot of improvements have been made for accessibility and will be available in the next version.

**Comment by marktaylor46** [comment]
Please note member was kind enough to leave some feedback here - some of that would suggest to me that there is still work to be done in this area?

http://www.rockettheme.com/forum/general-discussion/246086-accessibility-for-gantry-5

So should we raise a new issue so this get's addressed?

**Comment by w00fz** [comment]
Definitely. We considered this ticket for initial accessibility support.
We are still working on adding even more support for it as well as optimizing areas were needed. @kevlardev has been doing a sweet job on it!

Although now issues related to this should be opened in separated tickets, just to make it easier for us to track down and get more specific details about it.

Cheers!
